### DISCUSSION TITLE
Definition of  fun()

### DISCUSSION THREAD

**Discussion by lornawanjiru** [reach-sh/reach-lang#676] [Q&A] [answered]
const Player = {
  getHand: Fun([], UInt),
  seeOutcome: Fun([UInt], Null),
};

I would really appreciate if you could explain about the fun method. is it inbuilt? how do we define methods in reach.?

**Comment by jeapostrophe** [comment, marked as answer]
`Fun` is a type that describes a function, in the sample you include, it is describing something in the interface --- https://docs.reach.sh/rsh/compute/#rsh_Fun

You can define a function just like normal JavaScript --- https://docs.reach.sh/rsh/compute/#p_15 OR https://docs.reach.sh/rsh/compute/#arrow-expression
